enable test_memusage only if OS_LINUX is defined

#### Description of Problem:

The tests test_memusage.c uses [read_common_sizet(...)](https://github.com/OpenSCAP/openscap/blob/c8d3c87b49c5e90dbf9b087514e5a1f895d5a287/tests/API/probes/test_memusage.c#L16), but the method is available only [if OS_LINUX is defined](https://github.com/OpenSCAP/openscap/blob/c8d3c87b49c5e90dbf9b087514e5a1f895d5a287/src/common/memusage.c#L59) Otherwise the compilation of the test fails.

#### OpenSCAP Version:  [actual](https://github.com/OpenSCAP/openscap/tree/c8d3c87b49c5e90dbf9b087514e5a1f895d5a287)


#### Operating System & Version: reproduced on Solaris, but IMHO affects all non-linux OSes


#### Steps to Reproduce: `cmake && make`


#### Actual Results: 

```
...
[ 84%] Linking C executable test_memusage
ld: warning: file /usr/lib/amd64/libexslt.so: attempted multiple inclusion of file
ld: warning: file /usr/lib/64/libpcre.so: attempted multiple inclusion of file
ld: warning: file /usr/lib/64/libcurl.so: attempted multiple inclusion of file
Undefined                       first referenced
 symbol                             in file
read_common_sizet                   CMakeFiles/test_memusage.dir/test_memusage.c.o
ld: fatal: symbol referencing errors
collect2: error: ld returned 1 exit status
*** Error code 1
```

--- a/tests/API/probes/CMakeLists.txt
+++ b/tests/API/probes/CMakeLists.txt
@@ -39,11 +39,13 @@ target_include_directories(oval_fts_list PUBLIC
 target_link_libraries(oval_fts_list openscap)
 add_oscap_test("fts.sh")
 
-add_oscap_test_executable(test_memusage
-	"test_memusage.c"
-	"${CMAKE_SOURCE_DIR}/src/common/bfind.c"
-)
-target_include_directories(test_memusage PUBLIC
-	"${CMAKE_SOURCE_DIR}/src/common"
-)
-add_oscap_test("test_memusage.sh")
+if(OS_LINUX)
+	add_oscap_test_executable(test_memusage
+		"test_memusage.c"
+		"${CMAKE_SOURCE_DIR}/src/common/bfind.c"
+	)
+	target_include_directories(test_memusage PUBLIC
+		"${CMAKE_SOURCE_DIR}/src/common"
+	)
+	add_oscap_test("test_memusage.sh")
+endif(OS_LINUX)

